eZ/Publish/LegacySearch/RecommendationLegacySearchEngine.php

```diff
@@ -80,4 +80,60 @@ public function addNodeAssignment($mainNodeID, $objectID, $nodeAssignmentIDList,
             $this->legacySearchEngine->addNodeAssignment($mainNodeID, $objectID, $nodeAssignmentIDList, $isMoved);
         }
     }
+
+    public function cleanup()
+    {
+        if (method_exists($this->legacySearchEngine, 'cleanup')) {
+            return $this->legacySearchEngine->cleanup();
+        }
+    }
+
+    public function updateObjectsSection($objectIDs, $sectionID)
+    {
+        if (method_exists($this->legacySearchEngine, 'updateObjectsSection')) {
+            return $this->legacySearchEngine->updateObjectsSection($objectIDs, $sectionID);
+        }
+    }
+
+    public function updateObjectState($objectID, $objectStateList)
+    {
+        if (method_exists($this->legacySearchEngine, 'updateObjectState')) {
+            return $this->legacySearchEngine->updateObjectState($objectID, $objectStateList);
+        }
+    }
+
+    public function updateNodeSection($nodeID, $sectionID)
+    {
+        if (method_exists($this->legacySearchEngine, 'updateNodeSection')) {
+            return $this->legacySearchEngine->updateNodeSection($nodeID, $sectionID);
+        }
+    }
+
+    public function updateNodeVisibility($nodeID, $action)
+    {
+        if (method_exists($this->legacySearchEngine, 'updateNodeVisibility')) {
```

Comment:
I agree with this. It does no feel right to notify YooChoose service from LegacySearchEngine class, this should be handled in a different way IMHO.